src/io/fileops.c

```diff
@@ -423,7 +424,7 @@ MVMString * MVM_file_readlink(MVMThreadContext *tc, MVMString *path) {
 
     MVM_free(path_s);
     result = MVM_string_utf8_c8_decode(tc, tc->instance->VMString, req.ptr, strlen(req.ptr));
-    MVM_free(req.ptr);
+    uv__free(req.ptr);
```

Comment:
The change is better than what we had, but is it strictly correct?

The documentation at http://docs.libuv.org/en/v1.x/fs.html seems to suggest that the "correct" call would be `uv_fs_req_cleanup`

however

1) The documentation seems to suggest that one ought to be calling that function after every UV call that filled in a `uv_fs_t`
2) Looking at the implementation, there's a lot of C in it that we know will always end up as simply a call to `uv__free`

so if it's "correct" to call `uv_fs_req_cleanup` it sure looks like a lot of work to be seen to be doing the right thing. So I'm not sure if anything needs to change in this commit.

Reply:
Nice investigating. I suspect this isn't terribly hot code, so the extra work to be "correct" probably wouldn't hurt us too much. However, I'll leave it as is for now because I'm contemplating moving some of these not-directly-related-to-using-mimalloc changes out into their own PR(s).

Reply:
> I'm contemplating moving some of these not-directly-related-to-using-mimalloc changes out into their own PR(s).

I like this plan. I *almost* wrote something like "it would be useful to figure out a commit message that didn't mention mimalloc directly" because I thought it would be useful to cherry-pick just this commit. But, as you can sort of see from the previous sentence, I couldn't find a good way to phrase the suggestion, let alone the commit message...

Yes, I think that the "fixes for bugs you discovered" commits (at least) belong logically at the start of the chain of commits in this branch, but they would probably be better in their own PR(s) because we're going to apply them, however we end up implementing mimalloc (or getting delayed by problems)

Reply:
Just pulled a couple out, rebased, and force pushed.